File: client/src/app/educonnect/models/Course.ts
<<<<<<< HEAD
export class Course {
  courseId: number;
  courseName: string;
  description: string;
  teacherId: number;
  constructor(courseId: number, courseName: string, description: string, teacherId: number) {
    this.courseId = courseId;
    this.courseName = courseName;
    this.description = description;
    this.teacherId = teacherId;
  }
  logAttributes(): void {
    console.log('courseId:', this.courseId);
    console.log('courseName:', this.courseName);
    console.log('description:', this.description);
    console.log('teacherId:', this.teacherId);
  }
=======

export class Course {

>>>>>>> d76e0db293a8626f50b912ae8482fe15a23e1abc
}

File: server/src/main/java/com/edutech/progressive/entity/Course.java
package com.edutech.progressive.entity;

import com.fasterxml.jackson.annotation.JsonIgnore;

import javax.persistence.*;
import java.io.Serializable;

@Entity
@Table(name = "course")
public class Course implements Serializable {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "course_id")
    private int courseId;

    @Column(name = "course_name", nullable = false, length = 255)
    private String courseName;

    @Column(name = "description", length = 500)
    private String description;

    @Column(name = "teacher_id")
    private int teacherId;

    @JsonIgnore
    @ManyToOne(fetch = FetchType.LAZY, cascade = CascadeType.MERGE)
    @JoinColumn(name = "teacher_id", insertable = false, updatable = false)
    private Teacher teacher;

    public Course() {}

    public Course(int courseId, String courseName, String description, int teacherId) {
        this.courseId = courseId;
        this.courseName = courseName;
        this.description = description;
        this.teacherId = teacherId;
    }

    public int getCourseId() { return courseId; }
    public String getCourseName() { return courseName; }
    public String getDescription() { return description; }
    public int getTeacherId() { return teacherId; }
    public Teacher getTeacher() { return teacher; }

    public void setCourseId(int courseId) { this.courseId = courseId; }
    public void setCourseName(String courseName) { this.courseName = courseName; }
    public void setDescription(String description) { this.description = description; }
    public void setTeacherId(int teacherId) { this.teacherId = teacherId; }
    public void setTeacher(Teacher teacher) { this.teacher = teacher; }
}